xla/service/gpu/gpu_compiler.cc

```diff
@@ -2194,26 +2194,69 @@ absl::Status GpuCompiler::RunPostSchedulingPipelines(
   {
     HloPassPipeline pipeline("remat-pipeline");
 
-    HloCostAnalysis hlo_cost_analysis(ShapeSizeBytesFunction());
-    HloRematerialization::RematerializationModeConfig
-        rematerialization_mode_config(/*recompute=*/true, /*compress=*/true,
-                                      /*host_offload=*/false);
-    HloRematerialization::Options options(
-        hlo_cost_analysis, rematerialization_mode_config,
-        // Assume 75% of the total device memory is available for XLA.
-        /*memory_limit_bytes=*/scheduler_mem_limit,
-        /*block_size_limit=*/1, /*block_rematerialization_factor=*/1,
-        /*min_remat_size=*/0, /*compact_shape_function=*/nullptr,
-        /*host_memory_offload_config=*/std::nullopt);
-    HloRematerialization::RematerializationSizes sizes;
-    pipeline.AddPass<HloRematerialization>(options, sizes);
-    pipeline.AddPass<StreamAttributeAnnotator>();
-    pipeline.AddPass<OptimizationBarrierExpander>();
-
-    TF_ASSIGN_OR_RETURN(bool changed, pipeline.Run(module));
-    if (changed) {
-      VLOG(1) << "HloRematerialization saved "
-              << sizes.before_bytes - sizes.after_bytes << " bytes";
+    int64_t offloading_memory_limit_gb =
+        module->config().debug_options().xla_gpu_offloading_memory_limit_gb() *
+        1024 * 1024 * 1024;
+    float offloading_bandwidth =
+        module->config()
+            .debug_options()
+            .xla_gpu_offloading_bandwidth_bytes_per_sec();
+    float flops_per_sec =
+        module->config().debug_options().xla_gpu_offloading_flops_per_sec();
+    int64_t host_memory_space_color =
+        static_cast<int64_t>(se::MemoryType::kHost);
+
+    if (offloading_memory_limit_gb == 0) {
```

Comment:
There is a lot of duplicate code here. You should have one bit of code which builds the HloRematerialization::Options (based on the flags, etc.), then have just a single place where the passes are added to the pipeline.

Reply:
Thanks for pointing this out. It's changed accordingly.